### ISSUE TITLE
__IMPOSSIBLE__ triggered in TypeChecking.Substitute.piApply

### ISSUE THREAD

**Issue by asr** [agda/agda#1725]
From http://www.botik.ru/pub/local/Mechveliani/docon-A/0.03/docon-A-0.03.zip using the maint-2.4.2 branch:

``` bash
$ cd docon/source
$ agda -i <path_to_the_standard_library> Structures0.agda
...
piApply t = El {_getSort = Type (Max [Plus 1 (UnreducedLevel (Var 1 []))]), unEl = Sort (Type (Max [Plus 0 (UnreducedLevel (Var 1 []))]))}
  args = [[]r(Var 0 [Proj Structures0.UpMagma.magma])]
An internal error has occurred. Please report this as a bug.
Location of the error: src/full/Agda/TypeChecking/Substitute.hs:374
```

The file type-checks with Agda 2.4.2.4 and master.

`git bisect` points out  that issue was introduced by the commit 80625047ebbf0fbf8f338ef876712f84ed076853.

**Comment by asr** [comment]
Fixed by reverting 80625047ebbf0fbf8f338ef876712f84ed076853 in d5fd6f466a0ff8d543126c9015d5a860960d727c.

**Comment by asr** [comment]
TODO: Missing test case.

**Comment by andreasabel** [comment]
Ulf pointed out that he fixed #1701 in a different way, so the modifications to scope can be reverted.  However, instead of a git revert I would have just manually reverted the relevant 2 lines.  I can do this later.

**Comment by asr** [comment]
> However, instead of a git revert I would have just manually reverted the relevant 2 lines. 

You are right.

> I can do this later.

Ok.

**Comment by asr** [comment]
Testing the issue after 7a0838cedaeb0361c9f06e126c9d45c2341e6aa8...

**Comment by asr** [comment]
After 7a0838cedaeb0361c9f06e126c9d45c2341e6aa8 the issue continues fixed in maint-2.4.2 and master.
